crates/biome_js_analyze/src/lint/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,248 @@
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div onClick={()=>{})}></div>;
+    /// ```
+    /// ```jsx,expect_diagnostic
+    /// <span onClick={()=>{})}></span>;
+    /// ```
+    ///
+    /// When `<a>` does not have "href" attribute, that is non-interactive.
+    /// ```jsx,expect_diagnostic
+    /// <a onClick={()=>{})}></a>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <div role="button" onClick={()=>{})}></div>
+    /// <span role="link" onClick={()=>{})}></span>
+    /// <a href="http://example.com" onClick={()=>{})}></a>
+    /// ```
+    ///
+    pub NoStaticElementInteractions {
+        version: "next",
+        name: "noStaticElementInteractions",
+        language: "js",
+        sources: &[RuleSource::EslintJsxA11y("no-static-element-interactions")],
+        recommended: false,
+    }
+}
+
+// These are interactions defined by eslint-plugin-jsx-a11y.
+// ref: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/974275353598e9407c76bd4a50c331a953755cee/src/rules/no-static-element-interactions.js#L33-L37
+// ref: https://github.com/jsx-eslint/jsx-ast-utils/blob/main/src/eventHandlers.js
+lazy_static::lazy_static! {
+    static ref EVENT_TO_HANDLERS: FxHashMap<&'static str, Vec<&'static str>> = {
+        let mut m = FxHashMap::default();
+        m.insert("clipboard", vec!["onCopy", "onCut", "onPaste"]);
+        m.insert("composition", vec!["onCompositionEnd", "onCompositionStart", "onCompositionUpdate"]);
+        m.insert("keyboard", vec!["onKeyDown", "onKeyPress", "onKeyUp"]);
+        m.insert("focus", vec!["onFocus", "onBlur"]);
+        m.insert("form", vec!["onChange", "onInput", "onSubmit"]);
+        m.insert("mouse", vec![
+            "onClick", "onContextMenu", "onDblClick", "onDoubleClick", "onDrag", "onDragEnd",
+            "onDragEnter", "onDragExit", "onDragLeave", "onDragOver", "onDragStart", "onDrop",
+            "onMouseDown", "onMouseEnter", "onMouseLeave", "onMouseMove", "onMouseOut",
+            "onMouseOver", "onMouseUp"
+        ]);
+        m.insert("selection", vec!["onSelect"]);
+        m.insert("touch", vec!["onTouchCancel", "onTouchEnd", "onTouchMove", "onTouchStart"]);
+        m.insert("ui", vec!["onScroll"]);
+        m.insert("wheel", vec!["onWheel"]);
+        m.insert("media", vec![
+            "onAbort", "onCanPlay", "onCanPlayThrough", "onDurationChange", "onEmptied",
+            "onEncrypted", "onEnded", "onError", "onLoadedData", "onLoadedMetadata", "onLoadStart",
+            "onPause", "onPlay", "onPlaying", "onProgress", "onRateChange", "onSeeked", "onSeeking",
+            "onStalled", "onSuspend", "onTimeUpdate", "onVolumeChange", "onWaiting"
+        ]);
+        m.insert("image", vec!["onLoad", "onError"]);
+        m.insert("animation", vec!["onAnimationStart", "onAnimationEnd", "onAnimationIteration"]);
+        m.insert("transition", vec!["onTransitionEnd"]);
```

Comment:
Do we need elements (and handlers) other than in `CATEGORIES_TO_CHECK`?
We only use `focus`, `keyboard` and `mouse`.

Reply:
Thanks. So, I guess it means we don't have any other rules that need this `CATEGORIES_TO_CHECK` yet.

Alright, we can locate the hashmap here, but I suggest commenting out elements that aren't used here. It may be a waste of overhead. Insertion for `focus`, `keyboard`, and `mouse` is enough.

Reply:
@unvalley 
I see, I understand!
I'll fix this.

Reply:
@unvalley 
fixed in this https://github.com/biomejs/biome/commit/a9aa880a78e5128d92520c2ba14a06688470db38

Reply:
@unvalley 
Sorry, we are not able to addressed to additional comments, please leave them as they are.

Reply:
@ematipico@unvalley
Thank you for pointing this out.
As you said, it was causing overhead, so we defined it as a fixed-length array.

Also modified the comment to describe what is stored in this array and added the URL to the MDN reference describing each event handler

fixed in this commit [b330131](https://github.com/biomejs/biome/pull/2981/commits/b330131d44f547b234e54acd736c6ef496cf0b64)